department-of-veterans-affairs/vets-website
# Conditions-667/ Remove duplicate screen reader text in EditLink

## Are you removing, renaming or moving a folder in this PR?

Did you change site-wide styles, platform utilities or other infrastructure?
- [] No

## Summary
Noticed on our AT testing prototype which uses the ArrayBuilderCard that screen readers read out "Edit" twice when landing on the Edit link. "Edit edit dates"

In the ArrayBuilderCard code the `EditLink` component takes in the visually hidden `srText` prop. Screen readers are reading out the visible "Edit" text in the link along with the invisible `srText`, so it's announcing twice. The delete button is an actual button that uses the `srText` as an `aria-label`, so it doesn't have the same issue. Chose to remove the extra hidden span and just use the srText as the aria-label, matching the delete button implementation.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/667

## Testing done
- You can test this on any page that includes the ArrayBuilderCards, but you can test it on the new-conditions prototype without logging in.
- Go to http://localhost:3001/user-testing/new-conditions
- Click "Start your application without signing in"
- Click on "Cherry" to choose the Cherry demo. 
- Click "Continue" on the New Conditions page
- Add a condition on the Add a new condition page. Press continue
- On the Review your conditions page, turn on a screen reader
- Navigate to the Edit link.
- Listen. It should announce as "Edit {{condition}}" NOT "Edit edit {{condition}}"

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |<img width="1040" alt="Screenshot 2024-12-31 at 2 56 23 PM" src="https://github.com/user-attachments/assets/1d077ce3-4c01-481e-8d97-4a691f2628f6" />|       <img width="1040" alt="Screenshot 2024-12-31 at 2 54 54 PM" src="https://github.com/user-attachments/assets/202cac59-8cf6-4fd4-a077-6bb4e90d0f3a" />|


## What areas of the site does it impact?
Pages that use the ArrayBuilderCard component

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
